bin/ditto.js

```diff
@@ -25,19 +27,30 @@ const setupCommands = () => {
     .command('pull')
     .description('Sync copy from Ditto into working directory')
     .action(() => checkInit('pull'));
-  program
+    
+  const projectDescription = 'Add a Ditto project to sync copy from'
+  const projectCommand = program
     .command('project')
-    .description('Select Ditto project to sync copy from')
+    .description(projectDescription)
     .action(() => checkInit('project'));
+
+  projectCommand
+    .command('add')
+    .description(projectDescription)
+    .action(() => checkInit('project'));
+    
+  projectCommand
+    .command('remove')
+    .description('Stop syncing copy from a Ditto project')
+    .action(() => checkInit('project remove'));
 };
 
 const checkInit = async (command) => {
-  if (needsInit()) {
+  if (needsInit() && command !== 'project remove') {
```

Comment:
If the check isn't here, then if you run `project remove` with no projects in your config, the CLI will attempt to re-initialize by having you _add_ a project.

That behavior felt really counterintuitive to me (since it's a little strange to try and add a project in response to a removal command), so I moved the edge case up to avoid re-initializing if you're trying to remove a project, and instead just let the `removeProject()` run to detect that there are no projects and notify the user accordingly.

The alternative I considered was modifying the behavior of `init()` to account for this sort of edge case, but that would require passing down the action probably doing some much bigger refactoring, so this is the approach that I opted for. Super happy to consider other alternatives too!

Reply:
I also figured that it would be easy enough refactor and make this check more generic in the future if more commands came up that we wanted to avoid initializing with.